### PR TITLE
Refactor: Asset names and symbols

### PR DESCRIPTION
**Pull request by peronczyk**
The major changes are:

1. Using asset names on the left side of the transaction lists instead of the asset symbols
2. Using full asset names instead of just symbols in the top of the asset details page.
3. Consolidation of 2 separate transaction composables into one transactionData composable.
4. Major tweaks on how we pass the asset name and symbol to some components. For example some components were able to receive the tokens as property (ext-tokens) and also establish the tokens on its own. Now they are single purpose: so the parent needs to establish and pass the asset and the child purpose is to display them.
5. The logic that was establishing the transaction asset contractId was moved to middleware composable, so the transaction returned by the mdw is already "resolved" with correct data.
6. Some naming changes. Mostly because we use "token" as a reference to fungible tokens and "coin" as reference to the protocol main asset. If something can be both token and cain we need to call it an "asset".
7. Simplifying the AE coin symbol naming - instead of using both "Æ" and "AE" we are now using only "AE". So every protocol now has only one coin symbol without any quirky exceptions here and there.
8. Instead of displaying "AE Coin" label  for tokens we don't have any data we are now displaying "Unrecognized".

### CODE REVIEW

File: src/popup/components/Tokens.vue [hunk 1]
```diff
@@ -8,37 +8,43 @@
...
     <span class="symbols">
       <span
-        v-if="fromToken"
+        v-if="fromAsset"
         class="symbol"
-        v-text="truncateString(fromToken?.name ?? fromToken?.symbol)"
+        v-text="
+          truncateString(fullSymbol ? fromAsset?.name! : fromAsset?.symbol!)
+            || $t('common.unrecognized')
+        "
```
[the remark this answers is not in the document]
Comment (peronczyk) [reply src/popup/components/Tokens.vue:33]: Unfortunately it's not possible. This causes linting errors about wrong indentation :/ See for example `MultisigProposalDetails.vue` / line 239.

image

[the remark this answers is not in the document]
Comment (martinkaintas) [reply src/popup/components/Tokens.vue:33]: Yes, that's why I added the parenthesis too. If you add them I think you wont get the error

[the remark this answers is not in the document]
Comment (peronczyk) [reply src/popup/components/Tokens.vue:33]: You're right. Updated.

**Comment by CedrikNikita** [comment]
Actual Ae swap to WAE:
1) AE is marked as Unrecognized
2) Swap route is broken
image

**Comment by peronczyk** [comment]
> Actual Ae swap to WAE:
> 
> 1. AE is marked as Unrecognized
> 2. Swap route is broken

Fixed. Can you please take a look again?

**Comment by CedrikNikita** [comment]
> Fixed. Can you please take a look again?

swap routes

**Comment by Liubov-crypto** [comment]
> > Fixed. Can you please take a look again?
> 
> swap routes [ swap routes ](https://private-user-images.githubusercontent.com/9008074/310879811-71ae8c62-be46-408b-a5e5-66024f8c0ec5.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5MDk4NzcsIm5iZiI6MTcwOTkwOTU3NywicGF0aCI6Ii85MDA4MDc0LzMxMDg3OTgxMS03MWFlOGM2Mi1iZTQ2LTQwOGItYTVlNS02NjAyNGY4YzBlYzUuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDhUMTQ1MjU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTJkMWYyYjRlMDA0Zjg1MzNjZTlhNGQ5NTNmN2YzYmFlYWM5MTY0NjJmMTYyZDkzNjRkZGFiNWE0Njc2ZTRiNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.0ZzjDOuZxbHNOmfcoEQbyN5_ySV4iEVJVYNNYWkvWBs) [ ](https://private-user-images.githubusercontent.com/9008074/310879811-71ae8c62-be46-408b-a5e5-66024f8c0ec5.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5MDk4NzcsIm5iZiI6MTcwOTkwOTU3NywicGF0aCI6Ii85MDA4MDc0LzMxMDg3OTgxMS03MWFlOGM2Mi1iZTQ2LTQwOGItYTVlNS02NjAyNGY4YzBlYzUuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDhUMTQ1MjU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTJkMWYyYjRlMDA0Zjg1MzNjZTlhNGQ5NTNmN2YzYmFlYWM5MTY0NjJmMTYyZDkzNjRkZGFiNWE0Njc2ZTRiNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.0ZzjDOuZxbHNOmfcoEQbyN5_ySV4iEVJVYNNYWkvWBs)

it seems that for me this issue isn't reproducible:

dev

**Comment by CedrikNikita** [comment]
> it seems that for me this issue isn't reproducible:

My gif is based on wallet. superhero.com
.

File: src/protocols/aeternity/helpers/transactionTokenInfoResolvers.ts [hunk 2]
```diff
@@ -257,21 +263,23 @@ const swapExactTokensForAe = genLiquiditySwapResolver(
...
 );
 
 /**
  * ref: AedexV2Router.swap_ae_for_exact_tokens
  */
 const swapAeForExactTokens = genLiquiditySwapResolver(
-  ({ tx: { arguments: _arguments, amount } }) => ({
-    contractId: _arguments[1]?.value?.[0]?.value,
+  ({ tx: { amount } }) => ({
+    contractId: AE_CONTRACT_ID,
```
Comment (CedrikNikita) [inline src/protocols/aeternity/helpers/transactionTokenInfoResolvers.ts:279]: Why don't you use `...defaultCoin,` here instead?

Comment (peronczyk) [reply src/protocols/aeternity/helpers/transactionTokenInfoResolvers.ts]: Right, this was a leftover from my first approach. Updated...

**Comment by CedrikNikita** [comment]
image
The line that is coming from first token is broken.

**Comment by peronczyk** [comment]
> image The line that is coming from first token is broken.

Fixed. Can you please retest?

**Comment by Liubov-crypto** [comment]
For me the swap route looks good: 

swap

I have a question: is aeternity contract displayed properly? 
ae

also when I'm clicking to see the details of ae contact the redirection will be wrong: https://aescan.io/accounts/aeternity

Tx details: https://aescan.io/transactions/th_2fMGXFcCfyisc5Re4tRq6XrS2CLfzEfeskRpKxfuaP7DjyThWn

**Comment by CedrikNikita** [comment]
> For me the swap route looks good:
> 
> swap

Well first token suppose to Be `AE Coin`, so no, it is still not good. @Liubov-crypto can you please compare to a master branch behaviour, for swaps and pool movements.

**Comment by peronczyk** [comment]
> > For me the swap route looks good:
> Well first token suppose to Be `AE Coin`, so no, it is still not good. @Liubov-crypto can you please compare to a master branch behaviour, for swaps and pool movements.

Please try now @CedrikNikita @Liubov-crypto

**Comment by CedrikNikita** [comment]
> Please try now @CedrikNikita @Liubov-crypto

Swap route line

**Comment by Liubov-crypto** [comment]
Besides the issue that Nikita described I got the following: 

1. AE symbol is displaying as WAE in Tx details: 
Безіменний 117

2. AE contract is still broken, also check AE symbol near sum (here it is as Token): 




aepool




@peronczyk please check it.

**Comment by peronczyk** [comment]
@CedrikNikita 
Regarding the look of the swap route I had a conversation with Tsvetan and we agreed to change the design as follows:
image

@Liubov-crypto 
I've pushed new fixes for the mentioned bugs. I also removed the contract ID for the AE Coin for the pool transaction because the coin has no contract (it's not a token).

**Comment by Liubov-crypto** [comment]
LGTM. 

The issue with displaying WAE as AE I will log separately.

**Pull request merged**